src/pages/ReservationDetails.tsx

```diff
@@ -13,146 +26,259 @@ export default function ReservationApproval() {
   // gets reservationId from the URL params
   const { reservationId } = useParams();
   const [searchParams] = useSearchParams();
-  console.log(reservationId);
+  const confirmationToken = searchParams.get('confirmationToken');
 
-  // TODO: Load context from server!
-  const reservation: Reservation = {};
+  const [reservation, setReservation] = useState<Reservation>();
+  const [error, setError] = useState<string>();
+  const [confirmationButtonText, setConfirmationButtonText] = useState<string>(
+    'Confirm Reservation',
+  );
+  const [cancelButtonText, setCancelButtonText] =
+    useState<string>('Cancel Reservation');
+
+  const { reservationApi } = useContext(Context);
+  useEffect(() => {
+    reservationApi
+      .getReservation(reservationId ?? '0')
+      .then(val => {
+        setReservation(val);
+      })
+      .catch(e => {
+        if (e instanceof Response) {
+          e.json().then(j =>
+            (j.message as string).startsWith('Unable to find')
+              ? setError('Reservation not found!')
+              : setError(''),
+          );
+        }
+      });
+  }, []);
 
-  /**
-   * Checks if the given timestamp is tommorrow
-   * @param {number} timestamp the unix timestamp from reservation.time.from
-   * @return {boolean}
-   */
-  function isIn12Hours(timestamp: number): boolean {
+  const isIn12Hours = (timestamp: number): boolean => {
     const now = new Date();
     const difference = (timestamp - now.getTime()) * 60 * 60 * 1000;
     if (difference > 0 && difference <= 12) {
       return true;
     }
     return false;
-  }
+  };
+
+  const cancelReservation = async () => {
+    setCancelButtonText('...');
+    window.confirm('Are you sure you want to cancel your reservation?') &&
+      reservationApi
+        .deleteReservation(reservationId ?? '')
+        .then(() => {
+          setCancelButtonText('Reservation canceled!');
+        })
+        .catch(() => {
+          setCancelButtonText('Something went wrong!');
+          setError('Something went wrong canceling your reservation');
+        });
+  };
+
+  const confirmReservation = async () => {
+    setConfirmationButtonText('...');
+    await reservationApi
+      .putReservation(
+        { confirmed: true },
+        reservation?.id ?? '',
+        confirmationToken ?? '',
+      )
+      .then(() => {
+        setConfirmationButtonText('reservation confirmed!');
```

Comment:
Ich finde, hier sollten wir lieber so eine Box wie beim Fehler machen, nur in grün.